uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/calender-model_test.js

```diff
@@ -0,0 +1,61 @@
+define([
+  'mocks/data/work-pattern-data',
+  'leave-absences/shared/models/calender-model',
+], function (mockData) {
+  'use strict';
+
+  describe('Calender', function () {
+    var Calender,
+      WorkPatternAPI,
+      $q,
+      $rootScope;
+
+    beforeEach(module('leave-absences.models'));
+
+    beforeEach(inject(function (_Calender_, _WorkPatternAPI_, _$rootScope_, _$q_) {
+      Calender = _Calender_;
+      WorkPatternAPI = _WorkPatternAPI_;
+      $rootScope = _$rootScope_;
+      $q = _$q_;
+
+      spyOn(WorkPatternAPI, 'getCalendar').and.callThrough();
+    }));
+
+    afterEach(function () {
+      $rootScope.$apply();
+    });
+
+    describe('getCalendar()', function () {
+      var CalenderPromise,
+        deferred;
+
+      function commonSetUp(returnData) {
+        deferred = $q.defer();
+        WorkPatternAPI.getCalendar.and.returnValue(deferred.promise);
+        CalenderPromise = Calender.getCalendar();
+        deferred.resolve(returnData);
```

Comment:
can you group the lines of code so that they are easier to follow?
```js
deferred = $q.defer();
deferred.resolve(returnData);
WorkPatternAPI.getCalendar.and.returnValue(deferred.promise);

CalenderPromise = Calender.getCalendar();
```